app/services/html_parser.rb

```diff
@@ -71,40 +107,55 @@ def add_fullscreen_button_to_panel(html)
 
       codeblock.add_child(fullscreen_action)
     end
-    doc.to_html
+
+    @html = doc.to_html
+
+    self
   end
 
-  def wrap_all_tables(html)
-    doc = Nokogiri::HTML.fragment(html)
+  def wrap_all_tables
+    doc = Nokogiri::HTML.fragment(@html)
     doc.search("table").each { |table| table.swap("<div class='table-wrapper-paragraph'>#{table}</div>") }
-    doc.to_html
+    @html = doc.to_html
+
+    self
   end
 
-  def remove_empty_paragraphs(html)
-    doc = Nokogiri::HTML.fragment(html)
+  def remove_empty_paragraphs
+    doc = Nokogiri::HTML.fragment(@html)
     doc.css("p").select { |paragraph| all_children_are_blank?(paragraph) }.each(&:remove)
-    doc.to_html
+    @html = doc.to_html
+
+    self
   end
 
-  def escape_colon_emojis_in_codeblock(html)
-    html_doc = Nokogiri::HTML.fragment(html)
+  def escape_colon_emojis_in_codeblock
+    html_doc = Nokogiri::HTML.fragment(@html)
 
     html_doc.children.each do |el|
       next if el.name == "code"
 
       if el.search("code").empty?
-        el.swap(Html::ParseEmoji.call(el.to_html))
+        parsed_html = HtmlParser.new(el.to_html).parse_emojis.html
```

Comment:
why aren't we calling the parse emoji method directly instead of instancing a new instance of the same class we are in?

Reply:
That's a good question! Is there a way to do that under this setup?

Originally, this was using recursion and it called the method again. Since we removed the argument (`html`) from all the methods, I didn't see a way to really call the method itself and specify a new `html` argument. If we did that, we would overwrite the current value of `html` and it would return the incorrect data. In other words, we would have to do something like:

```ruby
@html = el.to_html
parsed_html = parse_emojis.html
el.swap(parsed_html)
```

I don't think this would work because it would reassign the value of `@html` to only a subset of the `html` from the loop, not the entire thing reformatted. Does that make sense?

Reply:
Ah got it! Maybe this "hack" is an indication that it should be its own separate service?

Reply:
I've had similar thoughts as well. This PR originally started on that path - making these separate services. It feels like the trick here is we sort of use these services/methods inconsistently. Some are called individually outside of the parsing process, some aren't. Breaking only a couple of these methods out into dedicated services feels a little strange to me too since we are supporting chaining and these are mostly used in the parsing funnel of logic.

I do understand what you mean though, there's a slight smell here. I'm definitely open to changing this up. I was leaning towards chalking that up to the fact this is legacy code and this PR still accomplishes the goal (move these to services) while making improvements against the current implementation (now support chaining, more spec coverage, etc.).